app/flame_chart/profile_model.ts

```diff
@@ -31,6 +31,15 @@ export type ThreadTimeline = {
   maxDepth: number;
 };
 
+type NumberedThreadName = {
```

Comment:
you can probably avoid creating this little struct and the parseNumberedThreadName function by using named capture groups:
https://javascript.info/regexp-groups#named-groups

Reply:
Ah nice, yeah that makes the change a bit smaller and also makes the regex less opaque :) done